integration-tests/reactive-messaging-rabbitmq-vault/src/test/java/io/quarkus/it/rabbitmq/RabbitMQConnectorDynCredsTest.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.it.rabbitmq;
+
+import static io.restassured.RestAssured.get;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.RabbitMQContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.vault.VaultContainer;
+
+import io.quarkus.it.rabbitmq.RabbitMQConnectorDynCredsTest.VaultResource;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.common.mapper.TypeRef;
+
+@QuarkusTest
+@QuarkusTestResource(VaultResource.class)
+public class RabbitMQConnectorDynCredsTest {
+
+    public static class VaultResource implements QuarkusTestResourceLifecycleManager {
+
+        RabbitMQContainer rabbit;
+        VaultContainer vault;
+
+        @Override
+        public Map<String, String> start() {
+            rabbit = new RabbitMQContainer(DockerImageName.parse("rabbitmq:3.9-management"))
```

Comment:
Would that work with dev services?

Reply:
No, see below for complete explanation.

Reply:
Scratch previous, explaining here...

To properly setup Vault I need Vault & RabbitMQ on the internal shared network and I need to know the RabbitMQ container is already initialized and healthy. This ordering is required because Vault verifies the RabbitMQ setup when setting the role configuration. I found no way to reliably do this without manually controlling the order myself.